src/test/java/com/github/checkstyle/patchfilter/SuppressionPatchFilterTest.java

```diff
@@ -51,33 +58,166 @@ public void testAccept() throws Exception {
     @Test
     public void testMultiChangesOnOneFileOne() throws Exception {
         final String fileName = getPath("MultiChangesOnOneFilePatch.txt");
-        final SuppressionPatchFilter filter = createSuppressionPatchFilter(fileName);
-        final LocalizedMessage message = new LocalizedMessage(4, 1, null, "msg", null,
-                SeverityLevel.ERROR, null, getClass(), null);
-        final AuditEvent ev = new AuditEvent(this, "Update.java", message);
-        assertTrue(filter.accept(ev),
-                "Audit event should be rejected when there are no matching patch filters");
+        final FileInputStream ins = new FileInputStream(fileName);
+        final UnifiedDiff diff = UnifiedDiffReader.parseUnifiedDiff(ins);
+        assertEquals(1, diff.getFiles().size());
+        final UnifiedDiffFile file1 = diff.getFiles().get(0);
+        assertNull(file1.getDiffCommand());
+        assertEquals("Origin.java", file1.getFromFile());
+        assertEquals("Update.java", file1.getToFile());
+        assertEquals(2, file1.getPatch().getDeltas().size());
+        assertNull(diff.getTail());
     }
 
     @Test
-    public void testMultiChangedFilesOnOnePatch() throws Exception {
+    @Ignore("https://github.com/java-diff-utils/java-diff-utils/issues/83")
+    public void testMultiChangedFilesOnOnePatchOne() throws Exception {
+        final String fileName = getPath("problem_diff_issue51.diff");
+        final FileInputStream ins = new FileInputStream(fileName);
+        final UnifiedDiff diff = UnifiedDiffReader.parseUnifiedDiff(ins);
+        assertEquals(2, diff.getFiles().size());
+        final UnifiedDiffFile file1 = diff.getFiles().get(0);
+        assertEquals("diff -U0 old/f1 new/f1", file1.getDiffCommand());
+        // here is f1 not old/f1 and new/f1
+        assertEquals("f1", file1.getFromFile());
+        assertEquals("f1", file1.getToFile());
```

Comment:
according to parse convention:
https://github.com/java-diff-utils/java-diff-utils/blob/a8e4ce2eff297c7700a6215bacb548e30151b948/java-diff-utils/src/main/java/com/github/difflib/unifieddiff/UnifiedDiffReader.java#L124-L130
there should be old/f1 and new/f1